File: OCT-Backend/src/Services/MediaInfoService/MediaInfo.Data/Entities/Participant.cs
using Enums.MediaInfoService;

namespace MediaInfo.Data.Entities;

public class Participant
{
    public Guid Id { get; set; }

    public string Name { get; set; }

    public string Surname { get; set; }

    public DateTime? Birthday { get; set; }

    public string Description { get; set; }

    public ParticipantRole Role { get; set; }

    public string Country { get; set; }
}

File: OCT-Front/src/app/models/media/participant.ts
import { Nullable } from "src/app/services/infrastructure/common";
import { CountryModel } from "../infrastructure/country";

export interface ParticipantModel {
    id: string;
    name: string;
    surname: string;
    birthday: Nullable<string>;
    description: string;
    role: ParticipantRole;
    country: string;
}

export interface ParticipantViewModel {
    id: string;
    name: string;
    surname: string;
    birthday: Nullable<string>;
    description: string;
    role: ParticipantRole;
    country: Nullable<CountryModel>;
}

export enum ParticipantRole {
    Actor,
    Producer
}

export interface ParticipantFilters {
    nameStartsWith: Nullable<string>;
    surnameStartsWith: Nullable<string>;
    role: Nullable<ParticipantRole>;
    limit: Nullable<number>;
    offset: Nullable<number>;
}

export interface UpdateParticipantModel {
    id: string;
    name: string;
    surname: string;
    birthday: Nullable<string>;
    description: string;
    role: ParticipantRole;
    country: string;
}